service/ws/server.go

```diff
@@ -52,14 +56,32 @@ func NewServer(cfg Config, log *mlog.Logger, opts ...Option) (*Server, error) {
 	return s, nil
 }
 
+// SendCh returns a channel that can be used to send messages to ws connections.
 func (s *Server) SendCh() chan<- Message {
 	return s.sendCh
 }
 
+// ReceiveCh returns a channel that can be used to receive messages from ws connections.
 func (s *Server) ReceiveCh() <-chan Message {
 	return s.receiveCh
 }
 
+// Close stops the websocket server and closes all the ws connections.
+// Must be called once all senders are done and cannot be called more than once.
```

Comment:
also sync.Once -able?

Reply:
It would but the assumption here is that the caller should not close it twice and if it does there's a problem somewhere. If we make it silently go through we'd never know. At this stage I'd prefer it to panic/fail test if we screw up something higher up. Of course if at some point we come up with a use-case for potentially closing it twice then by all means we should make use of something like `sync.Once`.